### PR TITLE
Importing only the icons we need

### PR DESCRIPTION
**Pull request by BoyWithSilverWings**
Removes dependency on the whole fontawesome css and icon fonts and moves to FA 5 with dynamic svg imports.

Allows custom icon components to be passed in by the user.

Relates to #37

### CODE REVIEW

**Comment by BoyWithSilverWings** [comment]
The datepicker components needs to be changed before we remove the fonts. 
But this requires that we change to a [custom input pattern](http://airbnb.io/react-dates/?path=/story/daypickersingledatecontroller--with-custom-input) for `react-dates`. Is that worth doing?
What do you think of the work so far? @soupette

**Comment by soupette** [comment]
> The datepicker components needs to be changed before we remove the fonts.
But this requires that we change to a custom input pattern for react-dates. Is that worth doing?

Can you explain me a bit more why do you think we need to change the datePicker?

**Comment by BoyWithSilverWings** [comment]
Currently the date picker uses a `::before` to show the calendar icon on the input that is placed by `react-dates`. When we work with `react-fontawesome` library, the icons are imported to the markup and cannot be used with psuedo elements. 
But we do not have control over the markup in current datepicker implmentation. If we move it to [rendering a custom input](http://airbnb.io/react-dates/?path=/story/daypickersingledatecontroller--with-custom-input) and then, triggering the library, the calendar icon can be inserted in place.

**Comment by soupette** [comment]
I see, sure let's do it! 

It will also be a good opportunity to migrate the code to react-hooks! 

Thanks a lot!

**Comment by BoyWithSilverWings** [comment]
Changed to custom input. Sticking to the class based format for this PR.
Removed font awesome fonts and other references.

File: buffet/package.json [hunk 1]
```diff
@@ -46,7 +46,10 @@
     "babel-core": "7.0.0-bridge.0"
   },
   "dependencies": {
-    "font-awesome": "^4.7.0",
+    "@fortawesome/fontawesome-svg-core": "^1.2.22",
```
Comment (Aurelsicoko) [inline buffet/package.json]: Why do we need all these icon libraries?

Comment (BoyWithSilverWings) [reply buffet/package.json]: When using the fontawesome css and fontawesome font as earlier, the user has to download the entire font and css even if he or she does not use any of them. 
When moving to these libraries, the clients who are using webpack, parcel or any of the other bundlers with tree shaking can have a bundle with only the icons we use without any extra baggage

**Comment by Aurelsicoko** [comment]
The icons are too big, we need to reduce the size a bit.

The pen and the bin are too big.
Screenshot 2019-09-18 at 15 34 55

The bin is too big.
Screenshot 2019-09-18 at 15 35 30

The eye is too big.
Screenshot 2019-09-18 at 15 36 14

When I say "too big", it means we should reduce the height, at least, 2px.

**Comment by BoyWithSilverWings** [comment]
@Aurelsicoko  Reduced the mentioned icon sizes

**Pull request merged**